erc1155/src/lib.rs

```diff
@@ -62,30 +62,30 @@ impl ERC1155Token {
 
 impl ERC1155TokenBase for ERC1155Token {
     fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
-        self.get_balance(account, id)
+        *self
+            .balances
+            .get(id)
+            .and_then(|m| m.get(account))
+            .unwrap_or(&0)
```

Comment:
`and_then` can map directly to a value:

```suggestion
        self
            .balances
            .get(id)
            .and_then(|m| *m.get(account))
            .unwrap_or(0)
```

I'm not insisting though